src/averageAges.js

```diff
@@ -38,6 +54,22 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const allChildren = people.map(children => children.mother);
+
+  const womenWithChildren = withChildren
+    ? women.filter(mother => (
+      allChildren.includes(mother.name)
+    ))
```

Comment:
`women.filter(mother => allChildren.includes(mother.name))` try to write it like this, i think it will work